be/src/vec/columns/column_impl.h

```diff
@@ -42,8 +42,8 @@ void IColumn::append_data_by_selector_impl(MutablePtr& res, const Selector& sele
                                "Size of selector: {} is larger than size of column: {}",
                                selector.size(), num_rows);
     }
-
-    res->reserve(num_rows);
+    DCHECK_GE(end, begin);
+    res->reserve(end - begin);
```

Comment:
add comment here to explain why need using this variable to reserve memory not using num rows